app/lib/tariff_synchronizer/taric_update.rb

```diff
@@ -31,14 +45,33 @@ def correct_sequence_pair?(next_update, previous_update)
         next_sequence[:sequence].to_i == expected_next_sequence
       end
 
-      def download(date)
-        TaricUpdateDownloader.new(date).perform
-      end
-
       def update_type
         :taric
       end
 
+      def applicable_updates
+        # Pull out the most recent update
+        current_update = most_recent_pending || most_recent_applied
+
+        # Bail if there isn't one (impossible from where the data is now)
+        return [] if current_update.blank?
+        # Bail if we need to correct with a manual rollback
+        return [] unless correct_filename_sequence?
+
+        accumulator = []
+
+        SEQUENCE_APPLICABLE_UPDATE_LIMIT.times do
+          current_update = new(
+            filename: current_update.next_update_sequence_update_filename,
+            issue_date: current_update.next_update_issue_date,
+          )
+
+          accumulator << current_update.dup
```

Comment:
it seems we do not need `.dup` in this case

Reply:
We should do it, yeah

Reply:
Removed. I don't think Sequel::Model implements initailize_dup or initialize_copy callbacks anyway so this basically does nothing useful. Just an end of day brain fart